src/analysis/lattice.h

```diff
@@ -0,0 +1,29 @@
+#ifndef wasm_analysis_lattice_h
+#define wasm_analysis_lattice_h
+
+#include "wasm.h"
+#include <bitset>
+#include <iostream>
+
+namespace wasm::analysis {
+
+enum LatticeComparison { NO_RELATION, EQUAL, LESS, GREATER };
+
+template<size_t N> struct BitsetPowersetLattice {
```

Comment:
Comments on each class would be helpful, I think. Just reading `BitsetPowerset` doesn't immediately suggest what that is, to me at least.

Reply:
Added some comments.